src/OrchardCore/OrchardCore.Users.Core/Services/UserService.cs

```diff
@@ -54,7 +54,7 @@ public async Task<IUser> AuthenticateAsync(string userName, string password, Act
                 return null;
             }
 
-            var user = await _userManager.FindByNameAsync(userName);
+            var user = await _userManager.FindByNameAsync(userName) ?? await _userManager.FindByEmailAsync(userName);
```

Comment:
Include this logic in `GetUserAsync()` instead and then:

```suggestion
            var user = await GetUserAsync(userName);
```